backend/lib/services/terminals/index.js

```diff
@@ -56,16 +56,47 @@ const TargetEnum = {
   SHOOT: 'shoot'
 }
 
-exports.create = function ({ user, namespace, name, target, body = {} }) {
+exports.create = function ({ user, coordinate: { namespace, name, target }, body = {} }) {
   return getOrCreateTerminalSession({ user, namespace, name, target, body })
 }
 
-exports.remove = function ({ user, namespace, name, target, body = {} }) {
-  return deleteTerminalSession({ user, namespace, name, target, body })
+exports.config = async function ({ user, coordinate: { namespace, name, target } }) {
+  return getTerminalConfig({ user, namespace, name, target })
 }
 
-exports.fetch = function ({ user, namespace, name, target, body = {} }) {
-  return fetchTerminalSession({ user, namespace, name, target, body })
+exports.list = function ({ user, coordinate: { namespace } }) {
+  return listTerminalSessions({ user, namespace })
+}
+
+exports.remove = function ({ user, body = {} }) {
+  return deleteTerminalSession({ user, body })
+}
+
+exports.fetch = function ({ user, body = {} }) {
+  return fetchTerminalSession({ user, body })
+}
+
+exports.heartbeat = async function ({ user, body = {} }) {
+  return heartbeatTerminalSession({ user, body })
+}
+
+function toTerminalMetadata (terminal) {
+  const metadata = _.pick(terminal.metadata, ['name', 'namespace'])
+  metadata.identifier = _.get(terminal, 'metadata.annotations["dashboard.gardener.cloud/identifier"]')
+  return metadata
+}
+
+function imageHelpText (terminal) {
+  const containerImage = terminal.spec.host.pod.containerImage
+  const imageDescriptions = getConfigValue('terminal.containerImageDescriptions')
+
+  const imageDescription = _
+    .chain(imageDescriptions)
+    .map(({ image, description }) => ({ image: new RegExp(image), description }))
```

Comment:
`fooImage:1.1.2` will match `fooImage:131.2` and `fooImage:1.122` Only consider as regexp if it contains a `.*` or starts and ends with `/`?
```js
return = _
    .chain(imageDescriptions)
    .find(({ image, description }) => {
      if (/\.\*/.test(image)) {
         return new RegExp(image).test(containerImage)
      } 
      return image === containerImage
    })
    .get('description')
    .value()
```

Reply:
good point! 👍

Reply:
✅